src/test/run-make/crate-attributes-with-cfg_attr/lib.rs

```diff
@@ -0,0 +1,13 @@
+// Copyright 2015 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![cfg_attr(foo, crate_type="lib")]
```

Comment:
I wasn't quite sure how to make a `run-pass` test, but I've never actually looked at the `auxiliary` folder before so I wasn't really aware of how to do that. I'll try that now.

> and then also have a binary which uses conditional compilation to build itself as a binary instead

What do you mean? Doesn't rustc assume it's a binary by default if not otherwise specified? I don't see how I can specify a file that would build as a library by default and override it to a binary with an attribute (since `--crate-type` on the command-line overrides `#![crate_type]`).